src/CodeGen_LLVM.cpp

```diff
@@ -474,6 +475,8 @@ MangledNames get_mangled_names(const LoweredFunc &f, const Target &target) {
 }  // namespace
 
 std::unique_ptr<llvm::Module> CodeGen_LLVM::compile(const Module &input) {
+    input_module = &input;
```

Comment:
Module doesn't have a default constructor so I assumed it was intended to be not nullable. I can make Module nullable, etc. but I prefer to resolve this later if it is ok with you. The pointer is also const, which is nice. We could move to having an Optional or Nullable type I guess.